### PR TITLE
Nn 5469 last updated

### PR DESCRIPTION
**Pull request by nagendramoj**

### CODE REVIEW

File: server/routes/dataInsights/pleasAndFindingsTab/pleasAndFindingsTabPage.ts [hunk 1]
```diff
@@ -35,6 +36,9 @@ export default class PleasAndFindingsTabPage {
     const agencyId: AgencyId = user.activeCaseLoadId
 
     const chartSettingMap = {}
+    const lastModifiedDate = getFullDate(
+      (await this.chartApiService.getLastModifiedChart(username, '5a')).lastModifiedDate.toString()
```
Comment (neilmendum) [inline server/routes/dataInsights/pleasAndFindingsTab/pleasAndFindingsTabPage.ts:40]: Is `.toString()` required here?

File: server/routes/dataInsights/protectedAndResponsivityCharacteristicsTab/protectedAndResponsivityCharacteristicsTabPage.ts [hunk 2]
```diff
@@ -57,6 +58,9 @@ export default class ProtectedAndResponsivityCharacteristicsTabPage {
     const characteristics: DropDownEntry[] = getUniqueItems(chartDetails2a.chartEntries as ChartEntryHorizontalBar[], {
       source: (row: ChartEntryHorizontalBar) => row.characteristic,
     })
+    const lastModifiedDate = getFullDate(
+      (await this.chartApiService.getLastModifiedChart(username, '2a')).lastModifiedDate.toString()
```
Comment (neilmendum) [inline server/routes/dataInsights/protectedAndResponsivityCharacteristicsTab/protectedAndResponsivityCharacteristicsTabPage.ts:62]: Is `.toString()` required here?

File: server/routes/dataInsights/punishmentsTab/punishmentsTabPage.ts [hunk 3]
```diff
@@ -43,6 +44,10 @@ export default class PunishmentsTabPage {
 
     const chartSettingMap = {}
 
+    const lastModifiedDate = getFullDate(
+      (await this.chartApiService.getLastModifiedChart(username, '4a')).lastModifiedDate.toString()
```
Comment (neilmendum) [inline server/routes/dataInsights/punishmentsTab/punishmentsTabPage.ts:48]: Is `.toString()` required here?

File: server/utils/utils.test.ts [hunk 4]
```diff
@@ -217,3 +218,25 @@ describe('convertSubmittedDateTimeToDateObject', () => {
...
+    const dateTime = '2023-08-24T15:30:00'
+    const format = 'MMMM D, YYYY h:mm a' // e.g. "August 24, 2023 3:30 pm"
+    const result = getFullDate(dateTime, format)
+    expect(result).toBe('August 24, 2023 3:30 pm')
+  })
+
+  test('should handle invalid date string gracefully', () => {
+    // Depending on the implementation of `getDateOrTime` this might throw an error or return a default value.
+    // Adjust the expected result accordingly.
+    const result = getFullDate(new Date().toString())
```
Comment (neilmendum) [inline server/utils/utils.test.ts:239]: Is `.toString()` required here?

File: server/routes/dataInsights/punishmentsTab/punishments.test.ts [hunk 5]
```diff
@@ -34,6 +34,12 @@ beforeEach(() => {
       chartEntries: [],
     })
   )
+  chartApiService.getLastModifiedChart.mockResolvedValue(
+    testData.chartLastUpdatedResult({
+      chartName: '4c',
```
Comment (neilmendum) [inline server/routes/dataInsights/punishmentsTab/punishments.test.ts:39]: In `punishmentsTabPage.ts` we pass `4a` so good to make this `chartName: '4a'` ?

File: server/routes/dataInsights/pleasAndFindingsTab/pleasAndFindings.test.ts [hunk 6]
```diff
@@ -20,6 +20,12 @@ beforeEach(() => {
       chartEntries: [],
     })
   )
+  chartApiService.getLastModifiedChart.mockResolvedValue(
+    testData.chartLastUpdatedResult({
+      chartName: '1a',
```
Comment (neilmendum) [inline server/routes/dataInsights/pleasAndFindingsTab/pleasAndFindings.test.ts:25]: In `pleasAndFindingsTabPage.ts` we pass `5a` so good to make this `chartName: '5a'` ?

File: server/routes/dataInsights/offenceTypeTab/offenceType.test.ts [hunk 7]
```diff
@@ -20,6 +20,12 @@ beforeEach(() => {
       chartEntries: [],
     })
   )
+  chartApiService.getLastModifiedChart.mockResolvedValue(
+    testData.chartLastUpdatedResult({
+      chartName: '3b',
```
Comment (neilmendum) [inline server/routes/dataInsights/offenceTypeTab/offenceType.test.ts:25]: In `offenceTypeTabPage.ts` we pass `3a` so good to make this `chartName: '3a'` ?

File: server/views/pages/dataInsights/totalsAdjudicationsAndLocationsTab.njk [hunk 8]
```diff
@@ -27,6 +27,12 @@
 
     {{ dataInsightsTabHeaderSingular(tabsOptions) }}
 
+    <div class="govuk-!-margin-bottom-6">
+     <p class="govuk-body" data-qa="last-da">Datasource: NOMIS <br/>
```
Comment (neilmendum) [inline server/views/pages/dataInsights/totalsAdjudicationsAndLocationsTab.njk:31]: Looks like the indentation here is only 1 space, should be 4?

File: server/utils/utils.test.ts (hunk 4 above)
Comment (nagendramoj) [reply server/utils/utils.test.ts:239]: Here we are testing when a new Date() with different format is sent, whether it throws Invalid Date or time message.

File: server/routes/dataInsights/punishmentsTab/punishmentsTabPage.ts (hunk 3 above)
Comment (nagendramoj) [reply server/routes/dataInsights/punishmentsTab/punishmentsTabPage.ts:48]: Done

**Pull request merged**